app/CivData.ts

```diff
@@ -186,7 +185,7 @@ class CivData {
                 [upgrades.bloodlines, upgrades.scaleBardingArmor, upgrades.chainBardingArmor],
                 {
                     infantry: [
-                        { name: "Civ bonus", atk: 0, rof: 33, hp: 0, ma: 0, pa: 0 }
+                        { name: "Civ bonus", atk: 0, rof: 25, hp: 0, ma: 0, pa: 0 }
```

Comment:
Nope, there's a mathematical gotcha in the code that _led_ to the tech tree error in the first place.

All bonuses are calculated by reducing the reload time (seconds per attack) by a certain amount. When that's converted to attack speed (attacks per second), the factor is inverted. For example, Japanese infantry "attack 25% faster" according to the original manual. That's because their _reload_ is multiplied by 3/4. However, the implication for _speed_ is that it gets multiplied by 4/3, and so we have the new TT saying 33%.

Same reason as Aztecs have buildings work 11% faster -- production times (seconds per unit) are all multiplied by 0.9, which means units per second are all multiplied by 1/0.9 ~= 1.11.

It's very silly and very annoying, but for our purposes here means the old TT values are correct.

Reply:
🤯 👍